scripts/generate_upgrade_syncset.py

```diff
@@ -0,0 +1,301 @@
+#!/bin/env python
+
+import argparse
+import datetime
+import json
+import logging
+import os
+import re
+import subprocess
+import sys
+import ruamel.yaml as yaml
+from ruamel.yaml.scalarstring import DoubleQuotedScalarString as dq
+
+CS = yaml.comments.CommentedSeq
+CM = yaml.comments.CommentedMap
+
+# Script to generate the cluster list
+CLUSTER_LIST_EXEC = 'get-cluster-list.sh'
+# SyncSet CR constants
+SS_API_VERSION = 'hive.openshift.io/v1'
+SS_NAME = 'osd-upgrade-config'
+# UpgradeConfig CR constants
+UC_NAME = 'osd-upgrade-config'
+UC_NAMESPACE = 'openshift-managed-upgrade-operator'
+UC_API_VERSION = 'upgrade.managed.openshift.io/v1alpha1'
+UC_PDB_DRAIN_TIMEOUT = 60
+UC_FORCE = False
+
+def get_cluster_info(cluster_ids, groups, subgroup, environment):
+    """
+    Returns a list of clusters matching the supplied group and subgroup
+    :param group: Cluster group
+    :param subgroup: Cluster subgroup
+    :return: list of tuple pairs of cluster ID and cluster name
```

Comment:
This is returning a dict, not a list of 2-tuples.

```suggestion
    Returns a mapping of clusters matching the supplied group and subgroup
    :param group: Cluster group
    :param subgroup: Cluster subgroup
    :return: dict, keyed by cluster ID, of cluster names
```